src/overlays/actors/ovl_En_Stone_heishi/z_en_stone_heishi.c

```diff
@@ -35,54 +39,473 @@ const ActorInit En_Stone_heishi_InitVars = {
+                     sEnStoneHeishiAnimations[this->animIndex], -10.0f);
+}
+
+void EnStoneheishi_HeadTowardsLink(EnStoneheishi* this) {
+    s16 yawDiff;
+    s32 absYawDiff;
+
+    yawDiff = this->actor.yawTowardsPlayer - this->actor.world.rot.y;
+
+    absYawDiff = ABS_ALT(yawDiff);
+
+    this->targetHeadPosX = 0;
+
+    if ((this->actor.xzDistToPlayer < 200.0f) && (absYawDiff < 0x4E20)) {
+        this->targetHeadPosX = this->actor.yawTowardsPlayer - this->actor.world.rot.y;
+        if (this->targetHeadPosX >= 0x2711) {
+            this->targetHeadPosX = 0x2710;
+        } else {
+            if (this->targetHeadPosX < -0x2710) {
+                this->targetHeadPosX = -0x2710;
+            }
+        }
+    }
+}
+
+void func_80BC94B0(EnStoneheishi* this) {
+    this->textIndex = 0;
+    if (gSaveContext.save.weekEventReg[0x29] & 0x40) { // After drinking bottle
+        EnStoneheishi_ChangeAnim(this, EN_STONE_HEISHI_CHEER_WITH_SPEAR);
+        this->textIndex = 8;
+        this->actor.flags &= ~ACTOR_FLAG_80;
+    } else { // Initial configuration
+        EnStoneheishi_ChangeAnim(this, EN_STONE_HEISHI_WAVE);
+        if (gSaveContext.save.weekEventReg[0x29] & 0x80) {
+            this->textIndex = 2;
+        }
+    }
+    this->actor.textId = sEnStoneHeishiTextIds[this->textIndex];
+    this->action = EN_STONE_ACTION_1;
+    this->actionFunc = func_80BC9560;
+}
+
+void func_80BC9560(EnStoneheishi* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
+    s16 yawDiff;
+    s32 absYawDiff;
+
+    if (Actor_ProcessTalkRequest(&this->actor, &play->state)) {
+        func_80BC9660(this);
+        return;
+    }
+
+    if ((!(gSaveContext.save.weekEventReg[0x29] & 0x40)) && (play->actorCtx.unk4 != 100)) {
+        this->actor.flags |= ACTOR_FLAG_8000000;
+        return;
+    }
+
+    SkelAnime_Update(&this->skelAnime);
+
+    this->actor.flags &= ~ACTOR_FLAG_8000000;
+    yawDiff = this->actor.yawTowardsPlayer - this->actor.world.rot.y;
+
+    absYawDiff = ABS_ALT(yawDiff);
+
+    if ((absYawDiff < 0x18F1) && ((s32)(player->stateFlags1 << 8) >= 0)) {
+        func_800B8614(&this->actor, play, 70.0f);
+    }
+}
+
+void func_80BC9660(EnStoneheishi* this) {
+    this->textIdSet = false;
+    this->action = EN_STONE_ACTION_2;
+    this->actionFunc = func_80BC9680;
+}
+
+void func_80BC9680(EnStoneheishi* this, PlayState* play) {
+    f32 currentFrame = this->skelAnime.curFrame;
+
+    if ((this->textIndex == 0) || (this->textIndex == 2)) {
+        if (this->animIndex != EN_STONE_HEISHI_SIT_AND_REACH) {
+            if (fabsf(this->headRotY - this->targetHeadPosY) < 50.0f) {
+                EnStoneheishi_ChangeAnim(this, EN_STONE_HEISHI_SIT_AND_REACH);
+            }
+            return;
+        }
+    } else if (this->textIndex == 3) {
+        if (this->animIndex != EN_STONE_HEISHI_WAVE) {
+            if ((this->timer == 0) && (fabsf((f32)(this->headRotY - this->targetHeadPosY)) < 50.0f)) {
```

Comment:
```suggestion
            if ((this->timer == 0) && (fabsf(this->headRotY - this->targetHeadPosY) < 50.0f)) {
```
Is this cast needed?